profiles/mpl_funcs.py

```diff
@@ -0,0 +1,116 @@
+from matplotlib.offsetbox import AnnotationBbox, OffsetImage
+
+from main_window.competition.comp_app import MAP_MARKER
+from main_window.device_manager import DeviceType
+from main_window.data_entry_id import DataEntryIds
+from main_window.main_app import MainApp
+from textwrap import wrap
+
+from main_window.mplwidget import MplWidget
+
+
+
+def receive_map(self, plot_widget: MplWidget, device:DeviceType, label_name:str) -> None:
+    """
+    Updates the UI when a new map is available for display
+    """
+    children = plot_widget.canvas.ax.get_children()
+    for c in children:
+        if isinstance(c, AnnotationBbox):
+            c.remove()
+
+    zoom, radius, map_image, mark = self.map_data.get_map_value()
+
+    # plotMap UI modification
+    plot_widget.canvas.ax.set_axis_off()
+    plot_widget.canvas.ax.set_ylim(map_image.shape[0], 0)
+    plot_widget.canvas.ax.set_xlim(0, map_image.shape[1])
+
+    # Removes pesky white boarder
+    plot_widget.canvas.fig.subplots_adjust(left=0, bottom=0, right=1, top=1, wspace=0, hspace=0)
+
+    if self.im:
+        # Required because plotting images over old ones creates memory leak
+        # NOTE: im.set_data() can also be used
+        self.im.remove()
+
+    self.im = plot_widget.canvas.ax.imshow(map_image)
+
+    # updateMark UI modification
+    for i in range(len(mark)):
+        annotation_box = AnnotationBbox(OffsetImage(MAP_MARKER), mark[i], frameon=False)
+        plot_widget.canvas.ax.add_artist(annotation_box)
+
+    plot_widget.canvas.draw()
+
+
+
+def receive_time_series(self, plot_widget: MplWidget, device:DeviceType, label_name:str) -> None:
+    """
+        Setup for plotting time series
+        Clear previous plot, change subplot settings
+    """
+    children = plot_widget.canvas.ax.get_children()
+
+    #clear children if still exist from last plot
+    for c in children:
+        if isinstance(c, AnnotationBbox):
+            c.remove()
+
+    plot_widget.canvas.fig.subplots_adjust(left=0.2, bottom=0.1, right=0.95, top=0.9, wspace=0, hspace=0)
+
+    if self.im:
+        # Required because plotting images over old ones creates memory leak
+        # NOTE: im.set_data() can also be used
+        self.im.remove()
+
+    self.im = None
+
+    if label_name == "Acceleration" and not plot_widget.showing_checkboxes:
+        plot_widget.show_checkboxes()
+
+
+    plot_widget.canvas.ax.set_axis_on()
+    plot_widget.canvas.ax.set_aspect('auto')
+
+    plot_time_series(self, plot_widget, device, label_name)
+
+
+def plot_time_series(self, plot_widget: MplWidget, device:DeviceType, label_name:str) -> None:
+    """
+            Updates the UI when time series data is available for display
+    """
+    plot_widget.canvas.ax.cla()
+    data_entry_id = self.rocket_profile.label_to_dataID[label_name]
+
+    if label_name == "Acceleration":
+        colors = ["Red", "Blue", "Green"]
```

Comment:
Looks like this doesn't match the ordering in accelwidget.py's accel_checkboxes, so checking acceleration Y or Z shows the incorrect colours